### PR TITLE
set user.properties.skipStudyPermissions for eda and non-eda differently

### PR DESCRIPTION
**Pull request by asizemore**
We'd like to have eda sites check permissions, while non-eda sites (currently live microbiome) skip this permissions check.

Related to a discussion in [web-study-data-access](https://github.com/VEuPathDB/web-study-data-access/pull/8)

Tested on my mbio site with live site setup. No failures :) Working on getting the eda version of the dev site tested.

I'm not confident this is the right spot for the change? I know `wdkService` comes in with a `getCurrentUser` method but no `getUser` method. I also have seen `wdkService.getUser` implemented elsewhere so it must get that method sometime. Looking forward to your feedback!

### CODE REVIEW

File: Client/src/wrapWdkService.js [hunk 1]
```diff
@@ -26,5 +26,14 @@ export default wdkService => ({
...
+  }),
+  getUser: () => {
+    let user = wdkService.getCurrentUser();
+    if (useEda) {
+      user.properties.skipStudyPermissions = false;
+    } else {
+      user.properties.skipStudyPermissions = true;
+    }
+    return user;
+  }
```
Comment (dmfalke) [inline Client/src/wrapWdkService.js:38]: `wdkService.getCurrentUser()` returns a `Promise`, which requires slightly different handling. Also, this method should also be named `getCurrentUser` (I misspoke in our original conversation`.

```suggestion
  getCurrentUser: async () => {
    let user = await wdkService.getCurrentUser();
    if (useEda) {
      user.properties.skipStudyPermissions = false;
    } else {
      user.properties.skipStudyPermissions = true;
    }
    return user;
  }
```

**Review by dmfalke — CHANGES_REQUESTED** [review]
This is very close! See my suggestion.

File: Client/src/wrapWdkService.js (hunk 1 above)
Comment (asizemore) [reply Client/src/wrapWdkService.js]: Awesome! Thanks @dmfalke !

File: Client/src/wrapWdkService.js [hunk 2]
```diff
@@ -26,5 +26,14 @@ export default wdkService => ({
...
     path: '/site-messages'
-  })
+  }),
+  getCurrentUser: async () => {
+    let user = await wdkService.getCurrentUser();
+    if (useEda) {
+      user.properties.skipStudyPermissions = false;
+    } else {
+      user.properties.skipStudyPermissions = true;
+    }
```
Comment (jtlong3rd) [inline Client/src/wrapWdkService.js:36]:
```suggestion
    user.properties.skipStudyPermissions = !useEda;
```

**Review by jtlong3rd — APPROVED** [review]
Just one small suggestion from me!

File: Client/src/wrapWdkService.js (hunk 2 above)
Comment (asizemore) [reply Client/src/wrapWdkService.js]: ahh brilliant! Thanks @jtlong3rd !!

**Review by dmfalke — APPROVED** [review]

**Pull request merged**